### PR TITLE
man1: improve HTML formatting of man pages

### PR DESCRIPTION
**Pull request by garlick**
This adds sphinx markup to our section 1 manual pages to improve HTML rendering and cross referencing.  Most of this was already presented in #5520. 

This should be purely formatting changes, although some line breaks in the `.rst` files needed to be adjusted to avoid making lines too long.

### CODE REVIEW

File: doc/man1/flux-alloc.rst [hunk 1]
```diff
@@ -5,7 +5,6 @@
 flux-alloc(1)
 =============
 
-
```
Comment (grondo) [inline doc/man1/flux-alloc.rst:8]: commit message:
"cross-reference tags for options are relative" :arrow_right: "...options that are relative"?

**Review by grondo — APPROVED** [review]
Excellent! I appreciate the amount of work that went into this update and especially the thorough description of the changes and included references in the commit messages.

File: doc/man1/flux-alloc.rst (hunk 1 above)
Comment (garlick) [reply doc/man1/flux-alloc.rst:8]: Oh, actually it's the tags that are relative not the options.  I'll reword, it's an awkward sentence.

**Comment by garlick** [comment]
Thank you!  I fixed that commit message and I'll set MWP.

File: doc/man1/flux-alloc.rst (hunk 1 above)
Comment (grondo) [reply doc/man1/flux-alloc.rst:8]: Ah, now I see, sorry!

**Pull request merged**